### ISSUE TITLE
Implement `det_to_seg` pipeline stage

### ISSUE THREAD

**Issue by mkutu** [precision-sustainable-ag/agir-pipeline#26]
## Overview

Implement the `det_to_seg` stage following the same pattern as `jpg_to_det`. Takes
per-image YOLO `.txt` detection files and the original JPGs, runs SMP segmentation
inside each bounding box, and composites the results into a single full-image binary
mask PNG per image.

## Inputs

| Flag | Description |
|------|-------------|
| `--i` | Det artifacts dir (`artifacts/` from a `jpg_to_det` run, contains `.txt` files) |
| `--j` | Original JPGs dir |
| `--c` | Config YAML (includes model weights path) |
| `--o` | Output dir |
| `--device` | Torch device (e.g. `cuda:0`) |
| `--t` | Worker processes (0 = sequential) |
| `--fs` | Fail-stop |
| `--batch-id` | Auto-inferred from path if omitted |

Discovery is driven by `.txt` files in `--i`. For each `.txt`, find the matching `.jpg`
in `--j` by stem. Skip with a warning if no match.

## Output

One binary mask PNG per image composited from all detections.
Zero detections → write an all-zeros mask (still recorded as `ok`).

```
{output_dir}/
  det_to_seg/
    {run_id}/
      artifacts/
        masks/
          TX_1687344534_mask.png
      run_report.json
      manifest.json
      run.log
```

Manifest artifact key: `mask_path` (matches `docs/examples/manifest_example.json`).

The only stage-specific values the intern can't derive from looking at `jpg_to_det` are the manifest item shape and the `artifact_type` string. Worth adding a small section for those two things:

## Manifest & Report

Manifest ok item:

```json
{
  "image_id": "TX_1687344534",
  "status": "ok",
  "artifacts": { "mask_path": "masks/TX_1687344534_mask.png" },
  "checksum": { "mask_path": "sha256:..." },
  "size_bytes": { "mask_path": 182334 }
}
```

Report artifact type:

```python
report.add_artifact_type(
    artifact_type="segmentation_mask",
    path=str(artifacts_dir / "masks"),
    n_files=num_succeeded,
)
```

Everything else like timing, provenance, error list, exit codes, follows `jpg_to_det` directly.

### File layout

Mirror `jpg_to_det` exactly.

```
stages/det_to_seg/
├── __init__.py
├── cli.py
├── processor.py
├── segmentor.py
├── configs/default.yaml
├── tests/test_processor.py
└── pyproject.toml
```

### Error codes

Same as `jpg_to_det` plus one new code:

- `DET_READ_FAILED` — `.txt` missing, unreadable, or malformed
- `IMAGE_READ_FAILED`, `INFERENCE_FAILED`, `EXPORT_FAILED`,
  `MODEL_LOAD_FAILED`, `CFG_VALIDATION_FAILED`, `UNKNOWN`

### Config

```yaml
weights: /path/to/unet_mit_b4.pth
arch: Unet
encoder: mit_b4
threshold: 0.5
pad_divisor: 32
tile_size: 1024
overlap: 128
tiling: true
```

### `segmentor.py`

Based on the attached `infer_bbox.py`. Key adaptations:

1. **Replace JSON parsing with TXT parsing.** YOLO format: `cls xc yc w h conf`
   (normalized). Convert to absolute `xyxy` using image dims. No `cutout_id` needed.
2. **Replace `process_annotation()` with a compositing function.**
   Initialize `full_mask = np.zeros((H, W), dtype=np.uint8)`. For each bbox:
   crop → `predict_mask()` → `full_mask[y1:y2, x1:x2] |= mask_crop`.
   Write `full_mask * 255` as PNG.
3. **Drop all visualization outputs** (`make_overlay`, `make_triptych`, etc.).
4. **`DEVICE` global → `device` parameter** passed from `Processor`.

The inference functions (`predict_mask_single`, `predict_mask_tiled`, `predict_mask`,
tensor helpers, `build_seg_model`, `load_weights_flex`) are lifted from `infer_bbox.py`
unchanged.

### `processor.py`

`process_image()` steps, each in its own `try/except`:

1. Read JPG → `rgb` + dims `(H, W)`
2. Parse `.txt` → convert normalized boxes to absolute `xyxy`
3. Run compositing function → full-image mask
4. Write PNG to `artifacts/masks/{image_id}.png`

Follow the `ProcessPoolExecutor` + `_init_worker` / `_WORKER_MODEL` pattern from
`jpg_to_det` for parallel mode. Skip if mask already exists (idempotency).

### `pyproject.toml` deps

`segmentation-models-pytorch`, `timm`, `torch`, `torchvision`, `opencv-python`,
`numpy`, `pyyaml`

> `timm` must be listed explicitly — the `mit_b4` encoder requires it.

## GPU Script

Add `tests/gpu/det_to_seg_a100.sh` — Slurm script to run the stage on the A100
with `--device cuda:0` against a real batch. Mirror the existing `jpg_to_det`
GPU scripts.

## Reference
1. the inferencing approach this stage is based on.
    * [`infer_bbox.py`](https://github.com/user-attachments/files/26978874/infer_bbox.py)
2. Segmentation model located in JUNO at:
    * `/LTS/project/dash_agir/semifield-tools/models/plant_segmentor/PSA_15164_393_unets/train_2025-10-10_08_36_13/checkpoints/epoch=47-step=37200-val_loss=0.00.ckpt`
    * `arch=unet`
    * `encoder=mit_b4`
